### DISCUSSION TITLE
Beamforming of an antenna array

### DISCUSSION THREAD

**Discussion by vcyycv0** [NVlabs/sionna#509] [Q&A]
hello every

I want to emulate  the beamforming of an antenna array. How to control the phase and amplitude of the antenna array？

**Comment by jhoydis** [comment]
Hi @vcyycv0,

In link-level simulations, you would typically do this via precoding, e.g., https://nvlabs.github.io/sionna/api/ofdm.html#zfprecoder.
It is straight-forward to define you own precoded.

For coverage maps, you can also provide a precoding vector that applies a complex number to each antenna element, see https://nvlabs.github.io/sionna/api/rt.html#sionna.rt.Scene.coverage_map.

You might also want to have a look at #331 .

**Reply by vcyycv0** [reply]
thank you for your answer.
But I want to control the amplitude and phase of each element of the antenna array, before using Scene.compute_paths. Can Sionna provide this function or modify the amplitude and phase of a through code after using Scene.compute_paths and a, tau=paths.cir().
thanks again.

**Reply by jhoydis** [reply]
Typically you would want to apply the precoding after the computation of the channel impulse response as the two are independent. This is what is done in the first example I mentioned above: https://nvlabs.github.io/sionna/api/ofdm.html#zfprecoder
You are free to apply any precoding you want.
